go/performance/benchmarks/helpers.go

```diff
@@ -0,0 +1,199 @@
+// Copyright 2019 Liquidata, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"strconv"
+	"strings"
+
+	"github.com/liquidata-inc/dolt/go/libraries/doltcore/sql"
+	"github.com/liquidata-inc/dolt/go/store/types"
+)
+
+// Container is used to correctly format sql strings
+type Container struct {
+	c []string
+}
+
+// NewContainer creates a new SQLContainer
+func NewContainer(format string) *Container {
+	c := make([]string, 3)
+	switch format {
+	case sqlExt:
+		c[0] = "("
+		c[2] = ")"
+	case jsonExt:
+		c[0] = "{"
+		c[2] = "}"
+	default:
+		log.Fatalf("cannot create new container, unsupported format %s \n", format)
+	}
+	return &Container{c: c}
+}
+
+// InsertPayload returns the SQLContainer with the payload inserted, separated by the separator
+func (sc *Container) InsertPayload(payload []string, separator string) string {
+	sc.c[1] = strings.Join(payload, separator)
+	return strings.Join(sc.c, "")
+}
+
+func getColValue(row []string, colIndex int, col *SeedColumn, format string) string {
+	switch col.GenType {
+	case increment:
+		return genNomsTypeValueIncrement(row, colIndex, col, format)
+	case random:
+		return getNomsTypeValueRandom(col, format)
+	default:
+		log.Fatalf("cannot get column value, unsupported gen type %s \n", col.GenType)
+	}
+	return ""
+}
+
+func genNomsTypeValueIncrement(row []string, colIndex int, col *SeedColumn, format string) string {
+	switch col.Type {
+	case types.IntKind:
+		if len(row) > 0 {
+			old, err := strconv.Atoi(row[colIndex])
+			if err != nil {
+				log.Fatalf(err.Error())
+			}
+			return fmt.Sprintf("%d", old+1)
+		}
+		return "1"
+	default:
+		log.Fatalf("cannot generate incremental value, unsupported noms type %s \n", col.Type.String())
+	}
+	return ""
+}
+
+func getNomsTypeValueRandom(col *SeedColumn, format string) string {
+	switch col.Type {
+	case types.IntKind:
+		return fmt.Sprintf("%d", rand.Intn(1000))
```

Comment:
Future extension: I would probably make it so that dataset generation could take a seed for the random number generator, so that we can reproduce the dataset that was tested with.